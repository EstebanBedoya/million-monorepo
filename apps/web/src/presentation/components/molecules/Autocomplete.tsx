'use client';

import { useState, useRef, useEffect } from 'react';
import { Search, ChevronDown, Loader2 } from 'lucide-react';
import { Input } from '@/presentation/components/atoms/Input';
import { cn } from '@/utils/cn';

export interface AutocompleteOption {
  value: string;
  label: string;
  description?: string;
}

interface AutocompleteProps {
  options: AutocompleteOption[];
  value: string;
  onChange: (value: string) => void;
  onSearch?: (query: string) => void;
  placeholder?: string;
  emptyMessage?: string;
  loading?: boolean;
  error?: boolean;
  disabled?: boolean;
  className?: string;
}

export function Autocomplete({
  options,
  value,
  onChange,
  onSearch,
  placeholder = 'Search...',
  emptyMessage = 'No results found',
  loading = false,
  error = false,
  disabled = false,
  className,
}: AutocompleteProps) {
  const [isOpen, setIsOpen] = useState(false);
  const [searchQuery, setSearchQuery] = useState('');
  const [highlightedIndex, setHighlightedIndex] = useState(-1);
  const containerRef = useRef<HTMLDivElement>(null);
  const inputRef = useRef<HTMLInputElement>(null);

  const selectedOption = options.find((opt) => opt.value === value);
  const displayValue = selectedOption?.label || '';

  const filteredOptions = options.filter((option) =>
    option.label.toLowerCase().includes(searchQuery.toLowerCase())
  );

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (containerRef.current && !containerRef.current.contains(event.target as Node)) {
        setIsOpen(false);
        setSearchQuery('');
      }
    };

    document.addEventListener('mousedown', handleClickOutside);
    return () => document.removeEventListener('mousedown', handleClickOutside);
  }, []);

  useEffect(() => {
    if (onSearch && searchQuery) {
      const timeoutId = setTimeout(() => {
        onSearch(searchQuery);
      }, 300);
      return () => clearTimeout(timeoutId);
    }
  }, [searchQuery, onSearch]);

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const query = e.target.value;
    setSearchQuery(query);
    setIsOpen(true);
    setHighlightedIndex(-1);
  };

  const handleOptionClick = (option: AutocompleteOption) => {
    onChange(option.value);
    setIsOpen(false);
    setSearchQuery('');
    setHighlightedIndex(-1);
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (!isOpen && (e.key === 'ArrowDown' || e.key === 'Enter')) {
      setIsOpen(true);
      return;
    }

    if (!isOpen) return;

    switch (e.key) {
      case 'ArrowDown':
        e.preventDefault();
        setHighlightedIndex((prev) =>
          prev < filteredOptions.length - 1 ? prev + 1 : prev
        );
        break;
      case 'ArrowUp':
        e.preventDefault();
        setHighlightedIndex((prev) => (prev > 0 ? prev - 1 : -1));
        break;
      case 'Enter':
        e.preventDefault();
        if (highlightedIndex >= 0 && filteredOptions[highlightedIndex]) {
          handleOptionClick(filteredOptions[highlightedIndex]);
        }
        break;
      case 'Escape':
        setIsOpen(false);
        setSearchQuery('');
        setHighlightedIndex(-1);
        break;
    }
  };

  const handleFocus = () => {
    setIsOpen(true);
  };

  return (
    <div ref={containerRef} className={cn('relative', className)}>
      <div className="relative">
        <Input
          ref={inputRef}
          type="text"
          value={isOpen ? searchQuery : displayValue}
          onChange={handleInputChange}
          onKeyDown={handleKeyDown}
          onFocus={handleFocus}
          placeholder={placeholder}
          disabled={disabled}
          error={error}
          icon={loading ? <Loader2 className="w-4 h-4 animate-spin" /> : <Search className="w-4 h-4" />}
          className="pr-10"
        />
        <button
          type="button"
          onClick={() => setIsOpen(!isOpen)}
          disabled={disabled}
          className="absolute right-3 top-1/2 -translate-y-1/2 text-secondary hover:text-foreground transition-colors disabled:opacity-50"
        >
          <ChevronDown
            className={cn(
              'w-4 h-4 transition-transform',
              isOpen && 'transform rotate-180'
            )}
          />
        </button>
      </div>

      {isOpen && (
        <div className="absolute z-50 w-full mt-1 bg-card border border-border rounded-md shadow-lg max-h-60 overflow-auto">
          {loading ? (
            <div className="p-4 text-center text-secondary">
              <Loader2 className="w-5 h-5 animate-spin mx-auto mb-2" />
              <p className="text-sm">Loading...</p>
            </div>
          ) : filteredOptions.length === 0 ? (
            <div className="p-4 text-center text-secondary text-sm">
              {emptyMessage}
            </div>
          ) : (
            <ul className="py-1">
              {filteredOptions.map((option, index) => (
                <li key={option.value}>
                  <button
                    type="button"
                    onClick={() => handleOptionClick(option)}
                    className={cn(
                      'w-full px-4 py-2 text-left hover:bg-accent/10 transition-colors',
                      highlightedIndex === index && 'bg-accent/10',
                      value === option.value && 'bg-accent/20 font-medium'
                    )}
                  >
                    <div className="text-sm text-foreground">{option.label}</div>
                    {option.description && (
                      <div className="text-xs text-secondary mt-0.5">
                        {option.description}
                      </div>
                    )}
                  </button>
                </li>
              ))}
            </ul>
          )}
        </div>
      )}
    </div>
  );
}

