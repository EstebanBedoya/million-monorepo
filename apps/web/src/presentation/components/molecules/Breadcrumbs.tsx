'use client';

import Link from 'next/link';
import { ChevronRight } from 'lucide-react';
import { Icon } from '../atoms/Icon';

export interface BreadcrumbItem {
  label: string;
  href?: string;
}

export interface BreadcrumbsProps {
  items: BreadcrumbItem[];
  className?: string;
}

export const Breadcrumbs = ({ items, className }: BreadcrumbsProps) => {
  return (
    <nav 
      aria-label="Breadcrumb" 
      className={className}
    >
      <ol className="flex items-center space-x-2 flex-wrap">
        {items.map((item, index) => {
          const isLast = index === items.length - 1;
          
          return (
            <li key={index} className="flex items-center">
              {index > 0 && (
                <Icon 
                  icon={ChevronRight} 
                  size="sm" 
                  className="text-secondary mx-2" 
                  aria-hidden={true}
                />
              )}
              
              {isLast || !item.href ? (
                <span 
                  className="text-sm font-medium text-foreground"
                  aria-current={isLast ? 'page' : undefined}
                >
                  {item.label}
                </span>
              ) : (
                <Link 
                  href={item.href}
                  className="text-sm text-secondary hover:text-accent transition-colors"
                >
                  {item.label}
                </Link>
              )}
            </li>
          );
        })}
      </ol>
    </nav>
  );
};

