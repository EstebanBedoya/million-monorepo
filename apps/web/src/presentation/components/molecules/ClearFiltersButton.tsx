'use client';

import { X } from 'lucide-react';
import { Button } from '../atoms/Button';
import { Icon } from '../atoms/Icon';

export interface ClearFiltersButtonProps {
  onClick: () => void;
  className?: string;
}

export const ClearFiltersButton = ({ onClick, className }: ClearFiltersButtonProps) => {
  return (
    <Button
      variant="secondary"
      onClick={onClick}
      className={`w-full lg:w-auto ${className}`}
      aria-label="Clear all filters"
    >
      <Icon icon={X} size="sm" />
      Clear All Filters
    </Button>
  );
};
