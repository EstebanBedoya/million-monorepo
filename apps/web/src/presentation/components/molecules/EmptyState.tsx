'use client';

import { Home } from 'lucide-react';
import { Button } from '../atoms/Button';
import { Icon } from '../atoms/Icon';

export interface EmptyStateProps {
  title?: string;
  message?: string;
  onReset?: () => void;
  className?: string;
}

export function EmptyState({ 
  title = "No Properties Found",
  message = "We couldn't find any properties matching your criteria. Try adjusting your filters.",
  onReset,
  className
}: EmptyStateProps) {
  return (
    <div className={`flex flex-col items-center justify-center py-16 px-4 text-center ${className}`} role="status">
      {/* Icon */}
      <div className="mb-6 p-6 rounded-full bg-secondary/10">
        <Icon 
          icon={Home} 
          size="xl" 
          className="text-secondary"
          aria-hidden={true}
        />
      </div>

      {/* Content */}
      <h3 className="text-2xl font-semibold text-foreground mb-3">
        {title}
      </h3>
      <p className="text-secondary max-w-md mb-6">
        {message}
      </p>

      {/* Reset button */}
      {onReset && (
        <Button
          onClick={onReset}
          aria-label="Clear all filters"
        >
          Clear All Filters
        </Button>
      )}
    </div>
  );
}
