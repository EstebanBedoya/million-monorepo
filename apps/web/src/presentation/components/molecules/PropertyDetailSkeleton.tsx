'use client';

import { Skeleton } from '../atoms/Skeleton';

export const PropertyDetailSkeleton = () => {
  return (
    <div className="min-h-screen bg-background">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8">
        {/* Back button and breadcrumbs skeleton */}
        <div className="mb-6">
          <Skeleton className="h-10 w-20 mb-4" animation="wave" />
          <div className="flex items-center space-x-2">
            <Skeleton className="h-4 w-12" animation="wave" />
            <Skeleton className="h-4 w-2" animation="wave" />
            <Skeleton className="h-4 w-16" animation="wave" />
            <Skeleton className="h-4 w-2" animation="wave" />
            <Skeleton className="h-4 w-24" animation="wave" />
          </div>
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-3 gap-8">
          {/* Main content area */}
          <div className="lg:col-span-2 space-y-6">
            {/* Hero image skeleton */}
            <div className="relative h-[400px] md:h-[500px] rounded-lg overflow-hidden bg-secondary/10">
              <Skeleton className="w-full h-full" animation="wave" />
              {/* Property type badge skeleton */}
              <div className="absolute top-4 left-4">
                <Skeleton className="h-6 w-16 rounded-full" animation="wave" />
              </div>
            </div>

            {/* Property title and price skeleton */}
            <div>
              <Skeleton className="h-10 w-3/4 mb-4" animation="wave" />
              <div className="mb-6">
                <Skeleton className="h-8 w-32 mb-3" animation="wave" />
                <div className="flex items-start gap-2">
                  <Skeleton className="h-5 w-5 mt-1" animation="wave" />
                  <div className="flex-1">
                    <Skeleton className="h-4 w-full mb-1" animation="wave" />
                    <Skeleton className="h-4 w-1/2" animation="wave" />
                  </div>
                </div>
              </div>

              {/* Property details skeleton */}
              <div className="p-4 bg-card border border-border rounded-lg">
                <div className="grid grid-cols-2 md:grid-cols-4 gap-4">
                  <div className="flex items-center gap-2">
                    <Skeleton className="h-5 w-5" animation="wave" />
                    <Skeleton className="h-4 w-16" animation="wave" />
                  </div>
                  <div className="flex items-center gap-2">
                    <Skeleton className="h-5 w-5" animation="wave" />
                    <Skeleton className="h-4 w-16" animation="wave" />
                  </div>
                  <div className="flex items-center gap-2">
                    <Skeleton className="h-5 w-5" animation="wave" />
                    <Skeleton className="h-4 w-16" animation="wave" />
                  </div>
                  <div className="flex items-center gap-2">
                    <Skeleton className="h-5 w-5" animation="wave" />
                    <Skeleton className="h-4 w-16" animation="wave" />
                  </div>
                </div>
              </div>
            </div>

            {/* Description section skeleton */}
            <div className="card-elevated p-6">
              <Skeleton className="h-6 w-32 mb-4" animation="wave" />
              <div className="space-y-3">
                <Skeleton className="h-4 w-full" animation="wave" />
                <Skeleton className="h-4 w-full" animation="wave" />
                <Skeleton className="h-4 w-3/4" animation="wave" />
                <Skeleton className="h-4 w-5/6" animation="wave" />
              </div>
            </div>

            {/* Property details section skeleton */}
            <div className="card-elevated p-6">
              <Skeleton className="h-6 w-32 mb-4" animation="wave" />
              <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                {[1, 2, 3, 4, 5, 6].map((i) => (
                  <div key={i} className="border-b border-border pb-3">
                    <Skeleton className="h-3 w-20 mb-1" animation="wave" />
                    <Skeleton className="h-4 w-24" animation="wave" />
                  </div>
                ))}
              </div>
            </div>
          </div>

          {/* Sidebar skeleton */}
          <div className="lg:col-span-1">
            <div className="card-elevated p-6 sticky top-8">
              <div className="mb-6">
                <Skeleton className="h-8 w-32" animation="wave" />
              </div>

              <div className="space-y-4 mb-6">
                {[1, 2, 3].map((i) => (
                  <div key={i} className="flex items-center justify-between py-3 border-b border-border">
                    <Skeleton className="h-4 w-20" animation="wave" />
                    <Skeleton className="h-4 w-16" animation="wave" />
                  </div>
                ))}
              </div>

              <div className="space-y-3">
                <Skeleton className="h-12 w-full" animation="wave" />
                <Skeleton className="h-12 w-full" animation="wave" />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

