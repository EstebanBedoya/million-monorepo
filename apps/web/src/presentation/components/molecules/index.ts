export { SearchInput } from './SearchInput';
export { PriceRange } from './PriceRange';
export { PriceSuggestions } from './PriceSuggestions';
export { PropertyTypeSelect } from './PropertyTypeSelect';
export { PropertyDetails } from './PropertyDetails';
export { LocationInfo } from './LocationInfo';
export { ClearFiltersButton } from './ClearFiltersButton';
export { PageButton } from './PageButton';
export { PageNumberButton } from './PageNumberButton';
export { EmptyState } from './EmptyState';
export { PropertyCardSkeleton, PropertyListSkeleton } from './PropertyCardSkeleton';
export { ThemeToggle } from './ThemeToggle';
export { AsyncBoundary } from './AsyncBoundary';
export { ErrorState } from './ErrorState';
export { LoadingState } from './LoadingState';
export { ErrorSimulationPanel } from './ErrorSimulationPanel';
export { Breadcrumbs } from './Breadcrumbs';
export { PropertyDetailSkeleton } from './PropertyDetailSkeleton';
export { SkeletonDemo } from './SkeletonDemo';
