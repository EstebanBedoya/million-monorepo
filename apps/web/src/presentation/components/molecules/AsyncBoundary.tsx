'use client';

import { ReactNode } from 'react';
import { PropertyListSkeleton } from './PropertyCardSkeleton';
import { EmptyState } from './EmptyState';
import { ErrorState } from './ErrorState';

export interface AsyncBoundaryProps {
  children: ReactNode;
  loading?: boolean;
  error?: Error | string | null;
  isEmpty?: boolean;
  emptyTitle?: string;
  emptyMessage?: string;
  onRetry?: () => void;
  onClearFilters?: () => void;
  skeletonCount?: number;
  className?: string;
}

export function AsyncBoundary({
  children,
  loading = false,
  error = null,
  isEmpty = false,
  emptyTitle,
  emptyMessage,
  onRetry,
  onClearFilters,
  skeletonCount = 12,
  className
}: AsyncBoundaryProps) {
  // Show loading state
  if (loading) {
    return <PropertyListSkeleton count={skeletonCount} className={className} />;
  }

  // Show error state
  if (error) {
    return (
      <ErrorState 
        error={error}
        onRetry={onRetry}
        className={className}
      />
    );
  }

  // Show empty state
  if (isEmpty) {
    return (
      <EmptyState 
        title={emptyTitle}
        message={emptyMessage}
        onReset={onClearFilters}
        className={className}
      />
    );
  }

  // Show content
  return <>{children}</>;
}
