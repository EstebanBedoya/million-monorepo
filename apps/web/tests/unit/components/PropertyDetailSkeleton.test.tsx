import { render, screen } from '@testing-library/react';
import { PropertyDetailSkeleton } from '../../../src/presentation/components/molecules/PropertyDetailSkeleton';

describe('PropertyDetailSkeleton', () => {
  it('should render skeleton elements', () => {
    render(<PropertyDetailSkeleton />);

    // Check for loading status elements
    const loadingElements = screen.getAllByRole('status');
    expect(loadingElements.length).toBeGreaterThan(0);
  });

  it('should render back button skeleton', () => {
    render(<PropertyDetailSkeleton />);

    // The back button skeleton should be present
    const loadingElements = screen.getAllByRole('status');
    expect(loadingElements.length).toBeGreaterThan(0);
  });

  it('should render breadcrumbs skeleton', () => {
    render(<PropertyDetailSkeleton />);

    // Multiple skeleton elements for breadcrumbs
    const loadingElements = screen.getAllByRole('status');
    expect(loadingElements.length).toBeGreaterThan(5);
  });

  it('should render main content skeleton', () => {
    render(<PropertyDetailSkeleton />);

    // Should have skeleton for hero image, title, price, etc.
    const loadingElements = screen.getAllByRole('status');
    expect(loadingElements.length).toBeGreaterThan(10);
  });

  it('should render sidebar skeleton', () => {
    render(<PropertyDetailSkeleton />);

    // Should have skeleton for sidebar content
    const loadingElements = screen.getAllByRole('status');
    expect(loadingElements.length).toBeGreaterThan(15);
  });

  it('should have proper accessibility attributes', () => {
    render(<PropertyDetailSkeleton />);

    const loadingElements = screen.getAllByRole('status');
    loadingElements.forEach(element => {
      expect(element).toHaveAttribute('aria-label', 'Loading...');
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<PropertyDetailSkeleton />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
