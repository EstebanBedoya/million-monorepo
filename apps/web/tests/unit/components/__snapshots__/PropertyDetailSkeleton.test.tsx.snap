// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetailSkeleton should match snapshot 1`] = `
<div
  class="min-h-screen bg-background"
>
  <div
    class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8"
  >
    <div
      class="mb-6"
    >
      <div
        aria-label="Loading..."
        class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-10 w-20 mb-4"
        role="status"
      />
      <div
        class="flex items-center space-x-2"
      >
        <div
          aria-label="Loading..."
          class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-12"
          role="status"
        />
        <div
          aria-label="Loading..."
          class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-2"
          role="status"
        />
        <div
          aria-label="Loading..."
          class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
          role="status"
        />
        <div
          aria-label="Loading..."
          class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-2"
          role="status"
        />
        <div
          aria-label="Loading..."
          class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
          role="status"
        />
      </div>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-3 gap-8"
    >
      <div
        class="lg:col-span-2 space-y-6"
      >
        <div
          class="relative h-[400px] md:h-[500px] rounded-lg overflow-hidden bg-secondary/10"
        >
          <div
            aria-label="Loading..."
            class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] w-full h-full"
            role="status"
          />
          <div
            class="absolute top-4 left-4"
          >
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] animate-[shimmer_2s_infinite] h-6 w-16 rounded-full"
              role="status"
            />
          </div>
        </div>
        <div>
          <div
            aria-label="Loading..."
            class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-10 w-3/4 mb-4"
            role="status"
          />
          <div
            class="mb-6"
          >
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-8 w-32 mb-3"
              role="status"
            />
            <div
              class="flex items-start gap-2"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-5 w-5 mt-1"
                role="status"
              />
              <div
                class="flex-1"
              >
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-full mb-1"
                  role="status"
                />
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-1/2"
                  role="status"
                />
              </div>
            </div>
          </div>
          <div
            class="p-4 bg-card border border-border rounded-lg"
          >
            <div
              class="grid grid-cols-2 md:grid-cols-4 gap-4"
            >
              <div
                class="flex items-center gap-2"
              >
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-5 w-5"
                  role="status"
                />
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                  role="status"
                />
              </div>
              <div
                class="flex items-center gap-2"
              >
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-5 w-5"
                  role="status"
                />
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                  role="status"
                />
              </div>
              <div
                class="flex items-center gap-2"
              >
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-5 w-5"
                  role="status"
                />
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                  role="status"
                />
              </div>
              <div
                class="flex items-center gap-2"
              >
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-5 w-5"
                  role="status"
                />
                <div
                  aria-label="Loading..."
                  class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                  role="status"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="card-elevated p-6"
        >
          <div
            aria-label="Loading..."
            class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-6 w-32 mb-4"
            role="status"
          />
          <div
            class="space-y-3"
          >
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-full"
              role="status"
            />
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-full"
              role="status"
            />
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-3/4"
              role="status"
            />
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-5/6"
              role="status"
            />
          </div>
        </div>
        <div
          class="card-elevated p-6"
        >
          <div
            aria-label="Loading..."
            class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-6 w-32 mb-4"
            role="status"
          />
          <div
            class="grid grid-cols-1 md:grid-cols-2 gap-4"
          >
            <div
              class="border-b border-border pb-3"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-3 w-20 mb-1"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
                role="status"
              />
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-3 w-20 mb-1"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
                role="status"
              />
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-3 w-20 mb-1"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
                role="status"
              />
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-3 w-20 mb-1"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
                role="status"
              />
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-3 w-20 mb-1"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
                role="status"
              />
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-3 w-20 mb-1"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-24"
                role="status"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="lg:col-span-1"
      >
        <div
          class="card-elevated p-6 sticky top-8"
        >
          <div
            class="mb-6"
          >
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-8 w-32"
              role="status"
            />
          </div>
          <div
            class="space-y-4 mb-6"
          >
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-20"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                role="status"
              />
            </div>
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-20"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                role="status"
              />
            </div>
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-20"
                role="status"
              />
              <div
                aria-label="Loading..."
                class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-4 w-16"
                role="status"
              />
            </div>
          </div>
          <div
            class="space-y-3"
          >
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-12 w-full"
              role="status"
            />
            <div
              aria-label="Loading..."
              class="bg-gradient-to-r from-secondary/20 via-secondary/30 to-secondary/20 bg-[length:200%_100%] rounded-md animate-[shimmer_2s_infinite] h-12 w-full"
              role="status"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
